src/autoscaler/eventgenerator/generator/appEvaluationManager.go

```diff
@@ -42,44 +42,40 @@ func NewAppEvaluationManager(logger lager.Logger, evaluateInterval time.Duration
 		breakerConfig:    breakerConfig,
 		cooldownExpired:  map[string]int64{},
 		breakerLock:      &sync.RWMutex{},
-		cooldownLock:     &sync.Mutex{},
+		cooldownLock:     &sync.RWMutex{},
 	}, nil
 }
 
 func (a *AppEvaluationManager) getTriggers(policyMap map[string]*models.AppPolicy) map[string][]*models.Trigger {
 	if policyMap == nil {
 		return nil
 	}
-	triggersByType := make(map[string][]*models.Trigger)
+	triggersByApp := make(map[string][]*models.Trigger)
 	now := a.emClock.Now().UnixNano()
```

Comment:
It is better to put now into the for-loop as for-loop may take a little longer time to finish.